benchmarks/predictions.py

```diff
@@ -0,0 +1,86 @@
+from typing import TYPE_CHECKING
+
+from jax.config import config
+
+config.update("jax_enable_x64", True)
+import jax.numpy as jnp
+import jax.random as jr
+
+import gpjax as gpx
+
+if TYPE_CHECKING:
+    from gpjax.gps import ConjugatePosterior
+
+
+class Gaussian:
+    param_names = [
+        "n_test",
+        "n_dims",
+    ]
+    params = [[100, 200, 500, 1000, 2000, 3000], [1, 2, 5]]
+
+    def setup(self, n_test, n_dims):
+        key = jr.PRNGKey(123)
+        self.X = jr.normal(key=key, shape=(100, n_dims))
+        self.y = jnp.sin(self.X[:, :1])
+        self.data = gpx.Dataset(X=self.X, y=self.y)
+        kernel = gpx.kernels.RBF(active_dims=list(range(n_dims)))
+        meanf = gpx.mean_functions.Constant()
+        self.prior = gpx.Prior(kernel=kernel, mean_function=meanf)
+        self.likelihood = gpx.likelihoods.Gaussian(num_datapoints=self.data.n)
+        self.posterior: ConjugatePosterior = self.prior * self.likelihood
+        key, subkey = jr.split(key)
+        self.xtest = jr.normal(key=subkey, shape=(n_test, n_dims))
+
+    def time_predict(self, n_test, n_dims):
+        self.posterior.predict(test_inputs=self.xtest, train_data=self.data)
+
+
+class Bernoulli:
+    param_names = [
+        "n_test",
+        "n_dims",
+    ]
+    params = [[100, 200, 500, 1000, 2000, 3000], [1, 2, 5]]
+
+    def setup(self, n_test, n_dims):
+        key = jr.PRNGKey(123)
+        self.X = jr.normal(key=key, shape=(100, n_dims))
+        self.y = jnp.sin(self.X[:, :1])
+        self.y = jnp.array(jnp.where(self.y > 0, 1, 0), dtype=jnp.float64)
+        self.data = gpx.Dataset(X=self.X, y=self.y)
+        kernel = gpx.kernels.RBF(active_dims=list(range(n_dims)))
+        meanf = gpx.mean_functions.Constant()
+        self.prior = gpx.Prior(kernel=kernel, mean_function=meanf)
+        self.likelihood = gpx.likelihoods.Bernoulli(num_datapoints=self.data.n)
+        self.posterior: ConjugatePosterior = self.prior * self.likelihood
```

Comment:
Minor question - won't this be a `NonConjugatePosterior`? Also not sure why there is a type annotation for this, but I could be missing something.

Reply:
Ah, this is actually not an issue thanks to a PR @frazane merged in the time since this PR was created.